File: RapidFindReplace-Demo-CS/ViewModels/MainWindowViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;

namespace RapidFindReplace_Demo_CS.ViewModels
{
    class MainWindowViewModel
    {

        /*
         * Auto generate a list of demo Windows
         * */
        List<Type> windows = null;
        public List<Type> AvailableWindows
        {
            get
            {
                if (windows == null)
                {
                    var children = Assembly.GetExecutingAssembly().GetTypes().Where(t => typeof(System.Windows.Window).IsAssignableFrom(t));

                    windows = new List<Type>();
                    foreach (Type t in children)
                    {
                        if (t.Name != "MainWindow")
                            windows.Add(t);
                        if (t.Name == "Basic") SelectedWindow = t;
                    }
                }
                windows.Reverse();
                return windows;
            }
        }

        Type _SelectedWindow;
        public Type SelectedWindow
        {
            get
            {
                return _SelectedWindow;
            }
            set
            {
                _SelectedWindow = value;
            }
        }
    }
}

File: RapidFindReplace-Demo-VB/ViewModels/MainWindowViewModel.vb
Imports System.Reflection

Namespace ViewModels

    Public Class MainWindowViewModel
        Private windows As List(Of Type) = Nothing

        Public ReadOnly Property AvailableWindows As List(Of Type)
            Get
                If (windows Is Nothing) Then
                    Dim children = Assembly.GetExecutingAssembly().GetTypes().Where(Function(t) GetType(System.Windows.Window).IsAssignableFrom(t))
                    
                    windows = New List(Of Type)
                    For Each t As Type In children
                        If (t.Name <> "MainWindow") Then
                            windows.Add(t)
                        End If
                        If (t.Name = "Basic") Then
                            SelectedWindow = t
                        End If
                    Next
                End If

                Return windows
            End Get
        End Property

        Private _SelectedWindow As Type

        Public Property SelectedWindow As Type
            Get
                Return _SelectedWindow
            End Get
            Set(value As Type)
                _SelectedWindow = value
            End Set
        End Property

    End Class
End Namespace
